### PR TITLE
jira issues -> report to slack when initial fetch has finished

### PR DESCRIPTION
**Pull request by lwsanty**
close [[DEV-1547]]
related to [[DEV-1556]]

Signed-off-by: lwsanty <lwsanty@gmail.com>

[DEV-1547]: https://athenianco.atlassian.net/browse/DEV-1547
[DEV-1556]: https://athenianco.atlassian.net/browse/DEV-1556

### CODE REVIEW

File: jira_fetcher.go [hunk 1]
```diff
@@ -127,6 +145,10 @@ func (f *JiraFetcher) handleIssues(ctx context.Context, ev *types.FetchEvent) er
 		fetchedIssues = make(map[types.IssueID]struct{})
 	)
 
+	reportInfo := func(ctx context.Context, format string, args ...interface{}) {
+		report.Info(ctx, "acc: %v, "+format, ev.AccID, args)
```
Comment (dennwc) [inline jira_fetcher.go:149]: You should set AccID to the context instead. See how [github fetcher does it](https://github.com/athenianco/metadata/blob/master/github/types/ghtypes.go#L81).

The idea is that as soon as you have AccID, you set it to the context, and all logs via `report` will include it as a filterable JSON field.

File: jira_fetcher.go [hunk 2]
```diff
@@ -461,29 +487,29 @@ func extractCommenters(issue gojira.Issue) ([]*gojira.User, error) {
 	return result, nil
 }
 
-func (f *JiraFetcher) setupCreds(ctx context.Context, ev *types.FetchEvent) (context.Context, error) {
+func (f *JiraFetcher) setupCreds(ctx context.Context, ev *types.FetchEvent) (context.Context, *database.Installation, error) {
```
Comment (dennwc) [inline jira_fetcher.go:490]: having previous comment in mind, you can remove the second return and set everything to the context.

File: jira_fetcher.go [hunk 3]
```diff
@@ -495,6 +521,35 @@ func (f *JiraFetcher) handleIssueUpdate(ctx context.Context, acc types.AccID, is
...
+}
+
+func (f *JiraFetcher) handleSlack(ctx context.Context, ev *types.FetchEvent, fields map[string]string) error {
+	if f.slack == nil {
+		return nil
+	}
+
+	return f.slack.SendMessage(ctx, slack.Message{
+		Channel: f.slackChannel,
+		Title:   "Jira fetch " + string(ev.EventType) + "complete",
```
Comment (dennwc) [inline jira_fetcher.go:547]:
```suggestion
		Title:   "Jira fetch " + string(ev.EventType) + " complete",
```

Comment (dennwc) [reply jira_fetcher.go:547]: Also, the event ID might be quite long. Maybe we can pass org name here?

Comment (lwsanty) [reply jira_fetcher.go:547]: there's no event id here, EventType is either issue, project, user or whatever
org name will be a part of the message fields

File: jira_fetcher.go (hunk 1 above)
Comment (lwsanty) [reply jira_fetcher.go]: Done.

File: jira_fetcher.go (hunk 2 above)
Comment (lwsanty) [reply jira_fetcher.go]: Done.

File: jira_fetcher.go (hunk 3 above)
Comment (lwsanty) [reply jira_fetcher.go]: Done.

**Review by dennwc — APPROVED** [review]

**Pull request merged**